sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/UseUriInsteadOfString.cs

```diff
@@ -56,12 +63,27 @@ protected sealed override void Initialize(SonarAnalysisContext context)
                     var methodDeclaration = (MethodDeclarationSyntax)c.Node;
                     var methodSymbol = c.SemanticModel.GetDeclaredSymbol(methodDeclaration);
 
+                    if (methodSymbol.IsOverride)
+                    {
+                        return;
+                    }
+
                     if (methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                        !methodSymbol.IsOverride &&
                         NameContainsUri(methodSymbol.Name))
                     {
                         c.ReportDiagnostic(Diagnostic.Create(rule_S3995, methodDeclaration.ReturnType.GetLocation()));
                     }
+
+                    foreach (ParameterSyntax param in methodDeclaration.ParameterList.Parameters)
```

Comment:
I know you really insist on this but we really don't care about the type of variables so you can use var.